From 4d6ed8ee5ac8753b7aae411f495d70cb682f57c2 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Thu, 1 Nov 2012 15:44:39 +0200
Subject: [PATCH 03/53] ti-tscadc-dt: Create ti-tscadc-dt DT adapter device

omap_device is going private.

Move the ti-tscadc-dt adapter device to arch/arm/mach-omap2.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 arch/arm/mach-omap2/Makefile       |    1 +
 arch/arm/mach-omap2/ti-tscadc-dt.c |    8 +++++++-
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/Makefile b/arch/arm/mach-omap2/Makefile
index c000bd5..6d56841 100644
--- a/arch/arm/mach-omap2/Makefile
+++ b/arch/arm/mach-omap2/Makefile
@@ -219,6 +219,7 @@ endif
 # AM3XX Device Tree adapters for legacy drivers
 obj-$(CONFIG_SOC_AM33XX)		+= ti-tscadc-dt.o
 obj-$(CONFIG_SOC_AM33XX)		+= da8xx-dt.o
+obj-$(CONFIG_SOC_AM33XX)		+= ti-tscadc-dt.o
 
 # Specific board support
 obj-$(CONFIG_MACH_OMAP_GENERIC)		+= board-generic.o
diff --git a/arch/arm/mach-omap2/ti-tscadc-dt.c b/arch/arm/mach-omap2/ti-tscadc-dt.c
index 76d9e76..16a1943 100644
--- a/arch/arm/mach-omap2/ti-tscadc-dt.c
+++ b/arch/arm/mach-omap2/ti-tscadc-dt.c
@@ -33,6 +33,7 @@
 #include <linux/input/ti_am335x_tsc.h>
 #include <linux/platform_data/ti_am335x_adc.h>
 #include <linux/mfd/ti_am335x_tscadc.h>
+
 #include "clock.h"
 #include "omap_device.h"
 
@@ -128,7 +129,12 @@ static int __devinit ti_tscadc_dt_probe(struct platform_device *pdev)
 
 static int __devexit ti_tscadc_dt_remove(struct platform_device *pdev)
 {
-	return -EINVAL;	/* not supporting removal yet */
+	struct ti_tscadc_priv *priv = platform_get_drvdata(pdev);
+
+	if (priv && priv->tscadc_pdev)
+		platform_device_unregister(priv->tscadc_pdev);
+
+	return 0;
 }
 
 static struct platform_driver ti_tscadc_dt_driver = {
-- 
1.7.7.6

